src/sgm_stereo_node.cpp

```diff
@@ -165,16 +115,47 @@ void StereoSGMNode::computeSGMStereoDisparity(const sensor_msgs::ImageConstPtr&
                                               const image_geometry::StereoCameraModel& model,
                                               stereo_msgs::DisparityImage& disp_msg)
 {
+  // Get OpenCV Image Mats (grayscale)
   const auto l_image = cv_bridge::toCvShare(l_image_msg, sensor_msgs::image_encodings::MONO8)->image;
   const auto r_image = cv_bridge::toCvShare(r_image_msg, sensor_msgs::image_encodings::MONO8)->image;
 
+  cv::Mat left_gray_img = l_image.clone();
+  cv::Mat right_gray_img = r_image.clone();
+
+  // ------------------- Compute Mask for Textureless Left regions --------------
+  cv::Mat left_grad;
+  extractSobelVerticalEdges(left_gray_img, "Left Gray ", left_grad);
```

Comment:
This step should be shared with SGM, as it does something similar (if not exactly the same).
At least as a TODO here.

Reply:
That's only necessary if I move this within SGM.